pkg/kubelet/dockertools/manager.go

```diff
@@ -824,7 +824,9 @@ func (dm *DockerManager) IsImagePresent(image kubecontainer.ImageSpec) (bool, er
 
 // Removes the specified image.
 func (dm *DockerManager) RemoveImage(image kubecontainer.ImageSpec) error {
-	return dm.client.RemoveImage(image.Image)
+	// TODO(harryz) currently Runtime interface does not provide other remove options.
+	_, err := dm.client.RemoveImage(image.Image, dockertypes.ImageRemoveOptions{})
```

Comment:
If you want the default docker behavior of "remove all the untagged images when the last tag is gone" you must set the `PruneChildren` bool argument to `true`.

See [here](https://github.com/docker/swarm/pull/1982/commits/1e22226b71c43f6698ce617ca3f544650091d61d#diff-a28a4b7a97d2368507318a8565e27ff2R481) for how I set it in docker/swarm.